scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+    mkdir -p "$output_folder"
+fi
+
+
+###############################################################################
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/k/generated/k_config/
+###############################################################################
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    kubeconfig="$arg_kubeconfig"
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    kubeconfig="$KUBECONFIG"
+else
+    kubeconfig="$HOME/.kube/config"
+fi
+echov -e "Using kubeconfig(s): ${COLOR_INFO}$kubeconfig${COLOR_NONE}"
+
+
+###############################################################################
+# Determine the list of contexts to search
+###############################################################################
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(k config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+
+echov "Validating contexts(s): "
+vc_n=0
+valid_context=()
+for context in "${contexts[@]}" ; do # for all contexts
+    [[ -z "$context" ]] && continue
+    if k --context $context get secrets -A > /dev/null 2>&1 ; then
+        echov -e "${COLOR_GREEN}\xE2\x9C\x94${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE}"
+        valid_context[vc_n]="$context"
+        vc_n=$((vc_n+1))
+    else
+        echov -e "${COLOR_RED}X${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE}"
+    fi
+done
+contexts=("${valid_context[@]}")
+
+if [[ -z "${contexts[@]}" ]] ; then
+    echoerr "no context(s) found in the kubeconfig file $kubeconfig!"
+    exit 1
+fi
+
+
+###############################################################################
+# Look for KubeStellar Helm chart
+###############################################################################
+echotitle "KubeStellar:"
+helm_context=""
+for context in "${contexts[@]}" ; do # for all contexts
+    chart="$(helm --kubeconfig $kubeconfig --kube-context $context list 2> /dev/null | grep 'core-chart' || true)"
+    if [[ "$chart" != "" ]] ; then
+        helm_context=$context
+        name="$(echo "$chart" | awk '{print $1}')" || true
+        version="$(echo "$chart" | awk '{print $NF}')" || true
+        secret="$(k --context $helm_context get secret --all-namespaces -l "owner=helm"  --no-headers -o name 2> /dev/null | grep "$name" || true)"
+        secret="${secret##*/}"
+        echo -e "- Helm chart ${COLOR_INFO}${name}${COLOR_NONE} (${COLOR_INFO}v${version}${COLOR_NONE}) in context ${COLOR_INFO}${context}${COLOR_NONE}"
+        echo -e "  - Secret=${COLOR_INFO}${secret}${COLOR_NONE}"
+        if [[ "$arg_yaml" == "true" ]] ; then
+            mkdir -p "$output_folder/kubestellar-core-chart"
+            echo -e "$(k --context $helm_context get secret $secret -o jsonpath='{.data.release}' | base64 -d | base64 -d | gzip -d | jq -r '.info'     | sed -e 's/\"/"/g')" > "$output_folder/kubestellar-core-chart/info.json"
```

Comment:
Why hack the JSON? Why use `echo -e`? Why not just do `k --context ... | jq '.info' > "$output_folder/kubestellar-core-chart/info.json"` ?

Reply:
Thank you for catching this.
It must be a leftover of when I was trying to display something for debug.